modules/distributing/slides-distributing.md

```diff
@@ -15,96 +15,465 @@ author: Jaro Camphuijsen, Lourens Veen
+
+### Distributing libraries
+
+- Make a package containing the library
+- Add metadata with dependencies
+- Upload the package to a repository
+
+note:
+The best way to distribute a library is by making a package for it, if there is a
+suitable packaging system available. Most modern programming languages have one, e.g.
+PyPI and pip for Python, or CRAN and install.packages() for R, or Cargo for Rust.
+
+So your steps here are to find the right packaging system based on the programming
+language the library is written in, then creating a package for that system (there is
+usually a manual/tutorial available), and uploading the package to the repository so
+that users can use their package manager to install it. Make sure to list all the
+dependencies in the metadata!
+
+For languages like Fortran or C++ that don't have a standard package manager, you could
+try Conda, or make packages for an operating system-specific packaging system like
+APT/DPKG (Debian/Ubuntu Linux), DNF/RPM (Red Hat Linux), HomeBrew or macPorts (macOS) or
+even EasyBuild, Spack or Nix (on High-Performance Computing)
+
+===
+
+<!-- .slide: data-state="standard" -->
+
+### Scenario 3: A whole application
+
+- Not so common in science
+- You have created a program
+- To be used by end users
+
+note:
+Scientists don't often make whole applications with user interfaces that anyone can just
+pick up and use. That's a lot of work, and you don't get credit for it in science.
+Nevertheless there are examples of this, e.g. MaxEnt in ecology, and it could happen. In
+this case, what you have is a program.
+
+===
+
+<!-- .slide: data-state="standard" -->
+
+### Distributing programs
+
+- In an app store!
+- As an installer?
+- As a package?
+- As an image?
+
+note:
+There are a few options for distributing a program. Packaging systems can be used,
+especially if the program is intended for use by programmers, or by more knowledgeable
+users. An App Store is basically a packaging system for applications, and that would be
+a good option too if available.
+
+Another option is to make an installer. That's a single file that is a working program,
+like a package, but it is also itself a program that when run will install your program
+onto the user's system. If you've installed software on Windows then you may have seen
+one, although they've mostly been replaced by app stores and my cause the operating
+system to warn about suspicious software.
+
+Finally, you could make an image here, but as before, it would be very large, and it
+would require special software on the user's computer, so in the here and now it's not a
+good option. It would be more resistant to changes in the operating system that break
+the program, because it includes the old version of the operating system. And then
+again, running an old operating system is a security risk.
+
+===
+
+<!-- .slide: data-state="standard" -->
+
+## Updating software
+
+- Everything changes
+- For scripts, you probably can
```

Comment:
?

Reply:
Yeah this is unclear. That line is the answer to the question "Can't I just put it online and be done with it?" but that's very unclear. I'll clarify.